### PR TITLE
Add Proxied Authorization (RFC 4370), Who Am I? (RFC 4532)

### PR DESCRIPTION
**Pull request by vetinari**
reworked #62 
- updated to use msgCtx

### CODE REVIEW

File: whoami_test.go [hunk 1]
```diff
@@ -0,0 +1,52 @@
+package ldap_test
+
+import (
+	"fmt"
+	"log"
+
+	"gopkg.in/ldap.v2"
+)
+
+func ExampleWhoAmI() {
```
Comment (johnweldon) [inline whoami_test.go]: Are there tests for `WhoAmI` possible?

Comment (vetinari) [inline whoami_test.go]: I'll add a encode / decode test to control_test.go

**Comment by vetinari** [comment]
... luckily the race fixed in #110 only happened in tip

**Comment by vetinari** [comment]
... waiting for #168 and #160

**Comment by johnweldon** [comment]
@vetinari I think this is the last one... any chance you still want to update this and we can get it merged in?

**Pull request closed without merging**